### PR TITLE
Collection usages in proxies are refactored.

### PR DESCRIPTION
**Pull request by sancar**
The places where HashSet is used converted to ArrayList.
The places where Collection returned to user replaced with
UnmodifiableLazyList.
Null checks used from ClientProxy replaced with Preconditions.

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/util/CollectionUtil.java [hunk 1]
```diff
@@ -92,4 +95,21 @@ public static boolean isNotEmpty(Collection collection) {
...
         return item;
     }
+
+    /**
+     * @param collection           collection of objects to be converted to collection of data
+     * @param serializationService will be used for converting object to data
+     * @return collection of data
+     * @throws java.lang.NullPointerException if collection is  empty, null, or contains a null element
+     */
+    public static <C> Collection<Data> objectToDataCollection(Collection<C> collection
```
Comment (ahmetmircik) [inline hazelcast/src/main/java/com/hazelcast/util/CollectionUtil.java]: needs to be tested in `CollectionUtilTest`.

File: hazelcast/src/main/java/com/hazelcast/util/CollectionUtil.java [hunk 2]
```diff
@@ -92,4 +95,21 @@ public static boolean isNotEmpty(Collection collection) {
         }
         return item;
     }
+
+    /**
+     * @param collection           collection of objects to be converted to collection of data
+     * @param serializationService will be used for converting object to data
+     * @return collection of data
+     * @throws java.lang.NullPointerException if collection is  empty, null, or contains a null element
```
Comment (ahmetmircik) [inline hazelcast/src/main/java/com/hazelcast/util/CollectionUtil.java]: seems there is no NPE when supplied collection is empty but javaDoc says opposite.

Comment (sancar) [inline hazelcast/src/main/java/com/hazelcast/util/CollectionUtil.java]: Fixing the javadoc

File: hazelcast/src/main/java/com/hazelcast/util/CollectionUtil.java (hunk 1 above)
Comment (sancar) [inline hazelcast/src/main/java/com/hazelcast/util/CollectionUtil.java]: I added the tests.

**Comment by ahmetmircik** [comment]
:+1:

**Comment by ihsandemir** [comment]
:+1:

**Pull request merged**